packages/medusa/src/models/batch-job.ts

```diff
@@ -1,30 +1,85 @@
-import { BeforeInsert, Column, Entity, PrimaryColumn } from "typeorm"
+import { AfterLoad, BeforeInsert, Column, Entity, JoinColumn, ManyToOne } from "typeorm"
 import { BatchJobStatus } from "../types/batch-job"
-import { DbAwareColumn } from "../utils/db-aware-column"
+import { DbAwareColumn, resolveDbType } from "../utils/db-aware-column"
 import { SoftDeletableEntity } from "../interfaces/models/soft-deletable-entity"
 import { generateEntityId } from "../utils/generate-entity-id"
+import { User } from "./user"
 
 @Entity()
 export class BatchJob extends SoftDeletableEntity {
   @DbAwareColumn({ type: "text" })
   type: string
 
-  @DbAwareColumn({ type: "enum", enum: BatchJobStatus })
-  status: BatchJobStatus
-
-  @Column({ type: "text", nullable: true })
+  @Column({ nullable: true })
   created_by: string | null
 
+  @ManyToOne(() => User)
+  @JoinColumn({ name: "created_by" })
+  created_by_user: User
+
   @DbAwareColumn({ type: "jsonb", nullable: true })
-  context: Record<string, unknown>
+  context: { retry_count?: number; max_retry?: number } & Record<string, unknown>
 
   @DbAwareColumn({ type: "jsonb", nullable: true })
   result: Record<string, unknown>
 
+  @Column({ type: "boolean", nullable: false, default: false })
+  dry_run: boolean = false;
+
+  @Column({ type: resolveDbType("timestamptz"), nullable: true })
+  pre_processing_at?: Date
+
+  @Column({ type: resolveDbType("timestamptz"), nullable: true })
+  processing_at?: Date
+
+  @Column({ type: resolveDbType("timestamptz"), nullable: true })
+  completed_at?: Date
+
+  @Column({ type: resolveDbType("timestamptz"), nullable: true })
+  canceled_at?: Date
+
+  @Column({ type: resolveDbType("timestamptz"), nullable: true })
+  ready_at?: Date
+
+  @Column({ type: resolveDbType("timestamptz"), nullable: true })
+  failed_at?: Date
```

Comment:
suggestion: 

Should we add an additional step to show when the user has confirmed:

So that this:
> - User creates batch job setting `created_at` -> status: created
> - Worker process begins work on batch job setting `pre_processing_at` -> status: pre_processing
> - Worker completes pre-processing on batch job setting `ready_at` -> status: ready
> - User confirms pre-processing results setting `processing_at` -> status: processing
> - Worker completes processing setting `completed_at` -> status: completed

Becomes:
- User creates batch job setting `created_at` -> status: created
- Worker process begins work on batch job setting `pre_processing_at` -> status: pre_processing
- Worker completes pre-processing on batch job setting `ready_at` -> status: ready
- **User confirms pre-processing results setting `confirmed_at` -> status: confirmed**
- Worker begins processing setting `processing_at` -> status: processing
- Worker completes processing setting `completed_at` -> status: completed

Reply:
Maybe it is worth considering renaming `ready` to something that gives a bit more context.

- I both like the old `awaiting_confirmation` because that is clear about the next action to take. 
- Also like `pre_processing_complete` as that describes where in the process we are.

Reply:
1. Yes this is exactly the event flow
2. I will do the update, but yeah those two status awaiting and confirm have been dropped 
3. Don’t think we need that status since a dry_run automatically need a confirmation, that is the difference. So basically, a dry_run job will no move to processing unless a user action has been made. Wdyt?

Reply:
If the job is dry run it could simply skip these statuses altogether i.e. confirmed_at would never be set it would go straight from "awaiting_confirmation_at"/"pre_processing_completed_at" to set "processing_at"?

This logic could be part of the `setReady` method?

Reply:
Yeah the setReady would
- set processing_at if not dry_run
- Don’t do anything if dry_run


Then the user call confirm at it will set the confirmed_at

the worker pick a job to process if
- dry_run and confirmed_at then process
- not dry_run then process

Reply:
On second thought it might be best to not give the worker the responsibility of having to think about `dry_run`.

So maybe `setReady` should do:
- if not dry_run -> set pre_processing_completed_at = confirmed_at = now
- if dry_run -> set pre_processing_completed_at = now

This way the worker would just have to wait for the confirmed event to be emitted from the BatchJobService?

Reply:
Iam agree on that, and that way we still know that it is a user confirmation or an internal confirmation just by looking at dry_run. 👌